test/Table_Tests/src/IO/Fetch_Spec.enso

```diff
@@ -94,3 +103,330 @@ add_specs suite_builder =
+            HTTP.clear_response_cache
+            expect_counts [0, 0] <|
+                Data.fetch url0 cache_policy=Cache_Policy.No_Cache
+                Data.fetch url1 cache_policy=Cache_Policy.No_Cache
+            expect_counts [0, 2] <|
+                Data.fetch url0 cache_policy=Cache_Policy.Use_Cache
+                Data.fetch url1 cache_policy=Cache_Policy.Use_Cache
+            HTTP.clear_response_cache
+            expect_counts [0, 2] <|
+                Data.fetch url0
+                Data.fetch url1
+
+        group_builder.specify "Should not cache Data.download" pending=pending_has_url <| Test.with_retries <|
+            target_file = enso_project.data / "transient" / "my_download0.txt"
+
+            HTTP.clear_response_cache
+            target_file.delete_if_exists
+
+            Data.download url0 target_file
+            get_num_response_cache_entries . should_equal 0
+
+            target_file.delete_if_exists
+
+        group_builder.specify "Data.download is not affected by caching limits" pending=pending_has_url <| Test.with_retries <|
+            target_file = enso_project.data / "transient" / "my_download0.txt"
+            Panic.with_finalizer reset_size_limits <|
+                EnsoHTTPResponseCache.setMaxTotalCacheSizeOverrideTestOnly 120
+                EnsoHTTPResponseCache.setMaxFileSizeOverrideTestOnly 100
+                Data.download base_url_with_slash+"test_download?length=200" target_file
+                target_file.read.length . should_equal 200
+                target_file.delete_if_exists
+
+        group_builder.specify "Should not cache for methods other than GET" pending=pending_has_url <| Test.with_retries <|
+            HTTP.clear_response_cache
+
+            expect_counts [0, 0] <|
+                Data.post url_post (Request_Body.Text "hello world")
+
+        group_builder.specify "HTTP request with a non-GET method should reject a Cache_Policy argument" pending=pending_has_url <| Test.with_retries <|
+            request = Request.new HTTP_Method.Post url_post [] Request_Body.Empty
+            HTTP.new.request request cache_policy=Cache_Policy.Use_Cache . should_fail_with Illegal_Argument
+
+        group_builder.specify "Should be able to clear caches" pending=pending_has_url <| Test.with_retries <|
+            HTTP.clear_response_cache
+            Data.fetch url0
+            get_num_response_cache_entries . should_equal 1
+            HTTP.clear_response_cache
+            get_num_response_cache_entries . should_equal 0
+
+        group_builder.specify "Cache key should only dependon the URI and headers" pending=pending_has_url <| Test.with_retries <|
+            HTTP.clear_response_cache
+            expect_counts [0, 1] <|
+                Data.fetch url0
+                Data.fetch url0
+            expect_counts [1, 2] <|
+                Data.fetch url1
+                Data.fetch url1
```

Comment:
The property defined in this test seems nice.

But I don't see how this test is checking this property in any more detail as many other tests. It feels completely redundant with the `"Cache should return the same repsonse"` test that checks the same things and a bit more.

Reply:
These were older tests I forgot to remove. I'll keep "Cache key should depend on the headers" though.